src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/CpsDiagnosticItemSource.cs

```diff
@@ -38,6 +41,12 @@ public CpsDiagnosticItemSource(Workspace workspace, string projectPath, ProjectI
                 {
                     Workspace.WorkspaceChanged += OnWorkspaceChangedLookForAnalyzer;
                     item.PropertyChanged += IVsHierarchyItem_PropertyChanged;
+
+                    // Now that we've subscribed, check once more in case we missed the event
+                    _analyzerReference ??= TryGetAnalyzerReference(Workspace.CurrentSolution);
+
+                    if (_analyzerReference != null)
+                        UnsubscribeFromEvents();
```

Comment:
I suspect a lock maybe needed here, imagine events of the following order

1. the second TryGetAnalyzerReference() above in the ctor is called on an old snapshot, which returns null
2. Project change event is fired, with new snapshot. and the handler below set `_analyzerReference` and get past the `if (_analyzerReference != null)` check
3. then the assignment in ctor set `_analyzerReference` to null.

Now we are unsubscribed to the workspace event

Reply:
@genlu Fixed; I did a force push of just the last commit. For sanity, I also updated the assignment in the WorkspaceChanged just in case.